src/mscorlib/src/System/Globalization/EncodingTable.Unix.cs

```diff
@@ -1,101 +1,162 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Collections;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Text;
+
 namespace System.Globalization
 {
-    using System;
-    using System.Text;
-    using System.Collections;
-    using System.Collections.Generic;
-    using System.Runtime.CompilerServices;
-    using System.Runtime.InteropServices;
-    using System.Runtime.Versioning;
-    using System.Security;
-    using System.Threading;
-    using System.Diagnostics.Contracts;
-
     internal static class EncodingTable
     {
+        //
+        // This caches the data item which is indexed by the code page value.
```

Comment:
Any reason why the data in `encodingDataTableItems` couldn't just be pre-allocated into a Dictionary, similar to the `encodingDataTable` Dictionary? Then there wouldn't be a need to lazily add items and wouldn't require any synchronization because it'd just be used for lookups.